src/_pytest/reports.py

```diff
@@ -272,3 +407,19 @@ def __init__(self, msg):
 
     def toterminal(self, out):
         out.line(self.longrepr, red=True)
+
+
+def pytest_report_serialize(report):
+    if isinstance(report, (TestReport, CollectReport)):
+        data = report._to_json()
+        data["_report_type"] = report.__class__.__name__
+        return data
+
+
+def pytest_report_unserialize(data):
+    if "_report_type" in data:
+        if data["_report_type"] == "TestReport":
+            return TestReport._from_json(data)
+        elif data["_report_type"] == "CollectReport":
+            return CollectReport._from_json(data)
+        assert "Unknown report_type unserialize data: {}".format(data["_report_type"])
```

Comment:
I think you mean `raise AssertionError(...)` or `assert False, ...` as it is this assertion is always true and is a noop

Reply:
So much for uncovered code.. :) - I've skipped it myself thinking that this is never meant to happen anyway..

Reply:
Ouch good catch. I will add a test for this so it gets covered. 👍

Reply:
Done, thanks again for catching this @asottile!